### DISCUSSION TITLE
where  (RuntimeSVGImporter) is  ?????

### DISCUSSION THREAD

**Discussion by forever12345678** [icosa-gallery/open-brush#531] [Q&A]
Please! I need help and I want to know where (RuntimeSVGImporter) is. 
Did it report an error? 
Don't have a namespace?

**Comment by andybak** [comment]
Are you saying:

"When I download the Unity Project and try and run it in the editor, I get an error about RuntimeSVGImporter"?

If so - can you give the exact error message AND any previous error messages you get. 

Also - how are you downloading the project? (git clone? download zip?)

If I had to guess I would say you haven't got git installed (or not installed properly). Several packages require that Unity can find a git executable on the system path. If this is the cause then it will be mentioned in one of the errors you get in the console.

**Reply by forever12345678** [reply]
Thank you for your reply.Thank you very much.
 i am Sorry, I don't speak English very well. 
I downloaded it from git clone, and I have downloaded and installed git。
My Unity couldn't load github dependencies, so I downloaded them from GitHub and added them from my disk. Then Unity told me that (RuntimeSVGImporter)  is no namespace for it. 
(RuntimeSVGImporter) in (ReferenceImage) script line 314 .
I think there might be a problem with my dependency package.
 I want to ask, is (RuntimeSVGImporter) from (com.unity.vectorgraphics)?

**Reply by Astronya** [reply]
I meet same issue too.Do you save the problem?Can I get your answer?

**Reply by andybak** [reply]
Unless by "same problem" you mean "every detail from @forever12345678 comments also apply to me" (which is unlikely...) then I'll need more detail.

The part where @forever12345678 said "My Unity couldn't load github dependencies, so I downloaded them from GitHub and added them from my disk." sounds suspect to me. That's not how you fix things. But I don't know what they meant by "couldn't load"

If you can explain your steps in detail with error messages in full then I can try and help.

**Reply by Astronya** [reply]
![微信图片_20240111144656](https://github.com/icosa-foundation/open-brush/assets/99575490/ae2da199-def8-48f8-8975-c54b7d54e08e)

That's the kind of error, the source code is like this. 

                // Try the cache first.
                m_FullSize = ImageCache.LoadImageCache(FilePath);
                if (m_FullSize == null)
                {
                    // TODO Move into the async code path?
                    var importer = new RuntimeSVGImporter();
                    m_FullSize = importer.ImportAsTexture(FilePath);
                    ImageCache.SaveImageCache(m_FullSize, FilePath);
                }
            }

I suspect this is because some libraries have been updated, preventing the RuntimeSVGImporter() function from being found.

**Comment by andybak** [comment]
Are you using the exact version of Unity specified in [ProjectVersion.txt](https://github.com/icosa-foundation/open-brush/blob/main/ProjectSettings/ProjectVersion.txt) ?

2021.3.30f1

**Reply by Astronya** [reply]
Yes I do.But I solved the problem, it was the wrong version of the dependency package. After using a matching version of the dependency package, it was resolved.
